### PR TITLE
feat(datasets): Release kedro-datasets 5.0.0

### PR DESCRIPTION
**Pull request by lrcouto**
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by deepyaman** [comment]
Just want to confirm—the intent is still to include the open PRs (#842, #812, other trivial open PRs) for release?

**Comment by lrcouto** [comment]
> Just want to confirm—the intent is still to include the open PRs (#842, #812, other trivial open PRs) for release?

Will confirm that later today.

**Comment by lrcouto** [comment]
> > Just want to confirm—the intent is still to include the open PRs (#842, #812, other trivial open PRs) for release?
> 
> Will confirm that later today.

It would be nice if we could at least include https://github.com/kedro-org/kedro-plugins/pull/812 on this release since it's another breaking change.

**Comment by deepyaman** [comment]
I'd argue #868 is also a trivial improvement that we should get in, unless there's a clear reason not to. I've pinged on #842 and will hopefully get the reviews soon; since it's gotten a number of user requests, and it's a new dataset (doesn't break any existing behavior), I would argue it could be done with fairly light review and get pretty immediate user feedback. 🤞 

Most of my other PRs can wait. :)

File: kedro-datasets/pyproject.toml [hunk 1]
```diff
@@ -193,7 +193,7 @@ rioxarray = ["kedro-datasets[rioxarray-geotiffdataset]"]
 
 # Docs requirements
 docs = [
-    "kedro-sphinx-theme==2024.4.0",
+    "kedro-sphinx-theme==2024.10.0",
```
Comment (ankatiyar) [inline kedro-datasets/pyproject.toml:196]:
```suggestion
    "kedro-sphinx-theme==2024.10.2",
```

**Comment by tdhooghe** [comment]
Happy to point PR #881 and update README.md and documentation accordingly, if you would want to merge it 🙂

**Review by ElenaKhaustova — APPROVED** [review]

**Comment by ElenaKhaustova** [comment]
> Happy to point PR #881 and update README.md and documentation accordingly, if you would want to merge it 🙂

This one will go to the next release 🙂

**Review by ravi-kumar-pilla — APPROVED** [review]

**Pull request merged**